lib/Protocol/FIX/Parser.pm

```diff
@@ -20,14 +20,14 @@ sub _parse_tag_pair {
     my ($protocol, $pair, $check_value) = @_;
     return unless $pair;
 
-    if ($pair =~ /^(\d{1,})($Protocol::FIX::TAG_SEPARATOR)(.*)$/) {
-        my ($tag_id, $value) = ($1, $3);
+    if ($pair =~ /^([0-9]+)$Protocol::FIX::TAG_SEPARATOR/op) {
```

Comment:
Note that the `/p` flag is a no-op since about perl 5.20 or so, but I *think* it's harmless there.